providers/default_rhel.rb

```diff
@@ -0,0 +1,172 @@
+#
+# Cookbook Name:: kernel-modules
+# Author:: Jeremy MAURO <j.mauro@criteo.com>
+#
+# Copyright 2016, Criteo.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+provides :kernel_module, platform_family: 'rhel'
+
+def whyrun_supported?
+  true
+end
+
+use_inline_resources
+
+action :save do
+  failing_sanity_check
+
+  # Requirements
+  package node['kernel_modules']['packages']
+
+  # Module init loading section
+  template modload_file do
+    source "#{node['platform_family']}-#{node['platform_version'].to_i}/init-load.erb"
+    cookbook 'kernel-modules'
+    mode '0755'
+    variables(
+      name: new_resource.name,
+      bin: modprobe_bin,
+    )
+  end
+
+  # Module loading options
+  resource_template_loading
+
+  resource_unloading(:save)
+
+  resource_load
+end
+
+action :load do
+  failing_sanity_check
+
+  # Module loading options
+  resource_template_loading(:load)
+
+  resource_unloading(:load)
+
+  resource_load(:load)
+end
+
+action :unload do
+  # First unload properly the module especially when the directive "remove"
+  # is used for the module
+  resource_unloading(:unload) if new_resource.reload
+
```

Comment:
Because if the module is used we might just want to wait the next reboot to take the change into account.